darts/dataprocessing/transformers/midas.py

```diff
@@ -0,0 +1,328 @@
+"""
+Mixed-data sampling (MIDAS) Transformer
+------------------
+"""
+from typing import Any, Dict, List, Mapping, Sequence, Union
+
+import numpy as np
+import pandas as pd
+from pandas import DatetimeIndex
+
+from darts import TimeSeries
+from darts.dataprocessing.transformers import (
+    FittableDataTransformer,
+    InvertibleDataTransformer,
+)
+from darts.logging import get_logger, raise_if, raise_if_not
+
+logger = get_logger(__name__)
+
+
+class MIDAS(FittableDataTransformer, InvertibleDataTransformer):
+    def __init__(
+        self,
+        rule: str,
```

Comment:
It will require a lot of changes since we heavily rely on the `pd.DataFrame.resample()` which require a `DatetimeIndex`. Possible but the whole logic would have to change. WDYT?

Reply:
Yes, we can do it in a future PR :)